src/mo_frontend/test_recovery.ml

```diff
@@ -256,9 +397,102 @@ actor Main {
     )
 
      with errors:
-    (unknown location): syntax error [M0001], unexpected token 'public'
-
-    (unknown location): syntax error [M0001], unexpected token '(' |}]
+    (unknown location): syntax error [M0001], unexpected token 'public', expected one of token or <phrase> sequence:
+      } (e.g. '}')
```

Comment:
Something has gone wrong - did you some revert the changes that @GoPavel  adopted from https://github.com/dfinity/motoko/pull/5108? We had managed to condense all the suggestions in a much shorter list.

Reply:
I discussed the matter with Pavel, it looks like those changes weren't merged into this PR. Instead, this was being developed here: https://github.com/serokell/motoko/pull/58

I've just merged them into this PR. Could you check whether it's fine now?